### ISSUE TITLE
How do I create a generator 

### ISSUE THREAD

**Issue by La-victoire** [geekcomputers/Python#1573]

**Comment by kumarvivek1752** [comment]
simply generator is a function but we write yield instead of return because when we write return in our function the function end on return. however, any of the local variables created during the function scope are destroyed and we can not accessible further. While in case of generator when it encounters a yield keyword the state of the function is frozen and all the variables are stored in memory until the generator is called again

basic syntax
```

def simpleGeneratorFun():
    yield 1
    yield 2
    yield 3
   
# x is a generator object
x = simpleGeneratorFun()
  
# Iterating over the generator object using next
print(x.next()) # In Python 3, __next__()
print(x.next())
print(x.next())
```

**Comment by Aliwahid17** [comment]
yes, you tell right about generators but instead of writing print statements multiple times use for or while loop

```python
def simpleGeneratorFun():
    yield 1
    yield 2
    yield 3
   
# x is a generator object
x = simpleGeneratorFun()
  
# Iterating over the generator object using next
[-]print(x.next()) # In Python 3, __next__()
[-]print(x.next())
[-]print(x.next())

[+]for i in range(3):
 [+]   print(x.next())
```

**Comment by Aliwahid17** [comment]
```
+ ji
```

**Comment by kumarvivek1752** [comment]
> yes, you tell right about generators but instead of writing print statements multiple times use for or while loop
> 
> ```python
> def simpleGeneratorFun():
>     yield 1
>     yield 2
>     yield 3
>    
> # x is a generator object
> x = simpleGeneratorFun()
>   
> # Iterating over the generator object using next
> print(x.next()) # In Python 3, __next__()
> print(x.next())
> print(x.next())
> 
>  for i in range(3):
>      print(x.next())
> ```

yeah but that is only for demonstration not for realworld use.

**Comment by Aliwahid17** [comment]
oh ok. by the way good demonstration

**Comment by Tevazu** [comment]
#### _Generators are objects used to create an iterable object in Python and do not take up memory space. For example, if we generate 100000 values and keep these values in memory, it will take up a lot of space, but this problem can be overcome thanks to generators. You can check this simple program which takes squares of the numbers 1, 2, 3, 4, 5 :_

```
def take_square():
    for i in range(1,6):
        yield (i ** 2)
```
#### _You can compare with the version which we didn't use generator:_
```
def take_square():
    result = []
    for i in range(1,6):
        result.append(i ** 2)
    return result
```

**Comment by Aliwahid17** [comment]
Yup thanks for telling. Now i know about it more

**Comment by sayampradhan** [comment]
### Generator for printing a string
```
#Creating generator
def generator():
    yield "SAYAM"
    yield "SAYAM"
    yield "SAYAM"

#Storing generator in a variable
generator = generator()

#Executing the generator
for value in generator:
    print(value)
```
